lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -429,17 +437,33 @@ void ELFImporter<T, HasAddend>::parseSymbols(object::ELFFile<T> &TheELF,
         or (Symbol.st_shndx == ELF::SHN_UNDEF))
       continue;
 
+    MetaAddress Address = MetaAddress::invalid();
     bool IsCode = Symbol.getType() == ELF::STT_FUNC;
-    if (!IsCode)
-      continue;
+    bool IsDataObject = Symbol.getType() == ELF::STT_OBJECT;
+    bool IsWeak = Symbol.getBinding() == ELF::STB_WEAK;
+    uint64_t Size = Symbol.st_size;
 
-    MetaAddress Address = MetaAddress::invalid();
-    Address = relocate(fromPC(Symbol.st_value));
-    auto It = Model->Functions.find(Address);
-    if (It == Model->Functions.end()) {
-      model::Function &Function = Model->Functions[Address];
-      if (MaybeName)
-        Function.OriginalName = *MaybeName;
+    if (IsCode)
+      Address = relocate(fromPC(Symbol.st_value));
+    else
+      Address = relocate(fromGeneric(Symbol.st_value));
+
+    if (IsCode) {
+      auto It = Model->Functions.find(Address);
+      if (It == Model->Functions.end()) {
+        model::Function &Function = Model->Functions[Address];
+        Function.Type = model::FunctionType::Regular;
+        if (MaybeName)
+          Function.OriginalName = *MaybeName;
+      }
+    }
```

Comment:
`else`

Reply:
`else` nothing, no? We only want to set `OriginalName` if `MaybeName` exists.